### DISCUSSION TITLE
Out of sort memory, consider increasing server sort buffer size (errno 1038)

### DISCUSSION THREAD

**Discussion by nishiyama-k0130** [planetscale/discussion#386] [Q&A] [answered]
Hello.

I faced 1038 error, that is lack of sort_buffer_size.

I am using as Hobby Plan.

When I throw query to planescale, return 1038 error.
So I tried to increase sort_buffer_size from 262144 to 1*1024*1024 by
```
set @@sort_buffer_size = 1*1024*1024;
```
It looked updated successfully, however could not. The value is returned to 262144 in other connection.

How can I solve this problem?

Any advice is helpful for me.

**Comment by orware** [comment, marked as answer]
Hello @nishiyama-k0130,

This is a known MySQL issue that will hopefully be resolved once the MySQL version used within PlanetScale is updated past 8.0.23.

The `sort_buffer_size` solution should work in the meanwhile for you but you'll have to make sure you set it in each MySQL session your application opens, since you cannot set it permanently. 

We currently recommend using the following in your sesion/connections if you are experiencing this error:

```sql
set sort_buffer_size=2097152;
```

If there is only one particular query triggering the issue you can issue the `SET` query right before it only, or if the issue is triggered by several queries you can run the `SET` query shortly after your connection is established to help ensure it is always increased for every connection.

Lastly, another suggestion that has also helped other users is that depending on the query this error is being triggered from, adding an index on the column you're sorting by may reduce the number of rows read in to memory during the sort process and reduce the occurrence of this error too.

**Reply by nishiyama-k0130** [reply]
Hello @orware 

Thank you for advice, I could solve my error by increase sort_buffer_size in my application connection.

Sincerely Regards,
